vim/vscode-insert.vim

```diff
@@ -25,7 +25,7 @@ augroup MultipleCursors
 augroup END
 
 " Multiple cursors support for visual line/block modes
-xnoremap <silent> ma :<C-u>call <SID>vscodePrepareMultipleCursors(1, 1)<CR>
-xnoremap <silent> mi :<C-u>call <SID>vscodePrepareMultipleCursors(0, 1)<CR>
-xnoremap <silent> mA :<C-u>call <SID>vscodePrepareMultipleCursors(1, 0)<CR>
-xnoremap <silent> mI :<C-u>call <SID>vscodePrepareMultipleCursors(0, 0)<CR>
+xnoremap <silent> ma <Cmd>call <SID>vscodePrepareMultipleCursors(1, 1)<CR>
+xnoremap <silent> mi <Cmd>call <SID>vscodePrepareMultipleCursors(0, 1)<CR>
+xnoremap <silent> mA <Cmd>call <SID>vscodePrepareMultipleCursors(1, 0)<CR>
+xnoremap <silent> mI <Cmd>call <SID>vscodePrepareMultipleCursors(0, 0)<CR>
```

Comment:
Tests fails for multiply cursors. You should probably edit `vscodePrepareMultipleCursors` fuction.

Reply:
Humm, so it looks like staying in visual mode causes the problem. I am not very experienced with vimscript, do you have any suggestions on how to exit visual mode, other than keeping the `:<C-U>`?

Reply:
This is because `s:vscodePrepareMultipleCursors` relies on `call feedkeys('i', 'n')`. But sending `i` will not produce insert mode from visual, you need to send `<Esc>` first. So, try to change `call feedkeys('i', 'n')` to `call feedkeys("\<Esc>i", 'n')` (you should use double quotes in this case).

Reply:
Yeah, you are right about the cause of the issue. However, the change you suggested didn't work. I tried a few combinations but I was not able to get it working, any other suggestions?

Reply:
Okay, let's go into insert mode first. Just call `i<Esc>` inside this mapping first and remove `feedkeys` from the function.

Reply:
Like `xnoremap <silent> ma <Esc>i<Cmd>call <SID>vscodePrepareMultipleCursors(1, 1)<CR>.

Reply:
Humm, so I got it to kind of work by leaving `call feedkeys('i', 'n')` (just `('i')` does the same thing too), and doing
```
xnoremap <silent> ma <Esc><Cmd>call <SID>vscodePrepareMultipleCursors(1, 1)<CR>
```
However, it is only working the first time and rarely after. I think there is timing issues going on with the cursors being spawned and going into insert mode.

Reply:
Could you try `xnoremap <silent> ma <Esc>i<Cmd>call <SID>vscodePrepareMultipleCursors(1, 1)<CR>` without `call feedkeys('i', 'n')`?

Reply:
I did, but that won't work, because `vscodePrepareMultipleCursors` needs to set some global flags *before* the insert mode autocommand is triggered.

Reply:
Oh, yes, you are right.
What if call it via `normal!`?

Reply:
I've tried a lot of things and none of them work all the time, even when I revert this PR.
I think this was caused by refactor, so I think we need to call on @asvetliakov . I think the cursor gets moved by neovim while entering insert mode, which steps on the vscode cursor spawning which happens before.

Reply:
Have you tried deleting the config? Multiply cursors work for me on master.

Reply:
Just tried solution with `feedkeys` and it also works. Let's check it on CI.

Reply:
I figured it out. Fixed :)